packages/flutter_driver/test/flutter_driver_test.dart

```diff
@@ -35,8 +35,8 @@ void main() {
       when(mockVM.isolates).thenReturn([mockIsolate]);
       when(mockIsolate.loadRunnable()).thenReturn(mockIsolate);
       when(mockIsolate.invokeExtension(any, any))
-          .thenReturn(new Future.value({'status': 'ok'}));
-      vmServiceConnectFunction = (_) => new Future.value(mockClient);
+          .thenReturn(new Future<Map<String, dynamic>>.value(<String, String>{'status': 'ok'}));
```

Comment:
I agree that JSON kind of makes this a mess, but in general doing this caught several cases where we were doing pointless work (e.g. returning values that were then ignored) so I'm not at all convinced that the cost is too high. I think the real problem is that we should try to turn the JSON data into structured data much more aggressively rather than shuttling the untyped JSON data around in futures and streams.